src/server/generic_family.cc

```diff
@@ -863,6 +863,99 @@ OpResult<uint64_t> OpTtl(Transaction* t, EngineShard* shard, string_view key) {
   }
 }
 
+ErrorReply RenameGeneric(CmdArgList args, bool destination_should_not_exist, Transaction* tx) {
+  string_view key[2] = {ArgS(args, 0), ArgS(args, 1)};
+
+  if (tx->GetUniqueShardCnt() == 1) {
+    tx->ReviveAutoJournal();  // Safe to use RENAME with single shard
+    auto cb = [&](Transaction* t, EngineShard* shard) {
+      return OpRen(t->GetOpArgs(shard), key[0], key[1], destination_should_not_exist);
+    };
+    OpResult<void> result = tx->ScheduleSingleHopT(std::move(cb));
+
+    return result.status();
+  }
+
+  Renamer renamer{tx, key[0], key[1], shard_set->size()};
+  return renamer.Rename(destination_should_not_exist);
+}
+
+void ExpireTimeGeneric(CmdArgList args, TimeUnit unit, Transaction* tx, SinkReplyBuilder* builder) {
+  string_view key = ArgS(args, 0);
+
+  auto cb = [&](Transaction* t, EngineShard* shard) { return OpExpireTime(t, shard, key); };
+  OpResult<uint64_t> result = tx->ScheduleSingleHopT(std::move(cb));
+
+  if (result) {
+    long ttl = (unit == TimeUnit::SEC) ? (result.value() + 500) / 1000 : result.value();
+    builder->SendLong(ttl);
+    return;
+  }
+
+  switch (result.status()) {
+    case OpStatus::KEY_NOTFOUND:
+      builder->SendLong(-2);
+      break;
+    default:
+      LOG_IF(ERROR, result.status() != OpStatus::SKIPPED)
+          << "Unexpected status " << result.status();
+      builder->SendLong(-1);
+      break;
+  }
+}
+
+void TtlGeneric(CmdArgList args, TimeUnit unit, Transaction* tx, SinkReplyBuilder* builder) {
+  string_view key = ArgS(args, 0);
+
+  auto cb = [&](Transaction* t, EngineShard* shard) { return OpTtl(t, shard, key); };
+  OpResult<uint64_t> result = tx->ScheduleSingleHopT(std::move(cb));
+
+  if (result) {
+    long ttl = (unit == TimeUnit::SEC) ? (result.value() + 500) / 1000 : result.value();
+    builder->SendLong(ttl);
+    return;
+  }
+
+  switch (result.status()) {
+    case OpStatus::KEY_NOTFOUND:
+      builder->SendLong(-2);
+      break;
+    default:
+      LOG_IF(ERROR, result.status() != OpStatus::SKIPPED)
+          << "Unexpected status " << result.status();
+      builder->SendLong(-1);
+      break;
+  }
+}
+
+std::optional<int32_t> ParseExpireOptionsOrReply(const CmdArgList args, SinkReplyBuilder* builder) {
+  int32_t flags = ExpireFlags::EXPIRE_ALWAYS;
+  for (auto& arg : args) {
+    string arg_sv = absl::AsciiStrToUpper(ToSV(arg));
+    if (arg_sv == "NX") {
+      flags |= ExpireFlags::EXPIRE_NX;
+    } else if (arg_sv == "XX") {
+      flags |= ExpireFlags::EXPIRE_XX;
+    } else if (arg_sv == "GT") {
+      flags |= ExpireFlags::EXPIRE_GT;
+    } else if (arg_sv == "LT") {
+      flags |= ExpireFlags::EXPIRE_LT;
+    } else {
+      builder->SendError(absl::StrCat("Unsupported option: ", arg_sv));
+      return nullopt;
+    }
```

Comment:
can you rewrite it with CmdArgParser

Reply:
I prefer to avoid rewriting stuff during refactoring PRs. this one is huge as it is.